src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -207,11 +439,68 @@ impl KVEngine for EtcdKVEngine {
         }
     }
 
+    /// Range get, return all key-value pairs start with prefix
     async fn range(&self, prefix: &KeyType) -> DatenLordResult<Vec<ValueType>> {
         let _timer = KV_METRICS.start_kv_operation_timer("range");
         let result = self.range_raw_key(prefix.to_string_key()).await?;
         Ok(result)
     }
+
+    /// Watch the key, return a receiver to receive the value
+    async fn watch(
+        &self,
+        prefix: &KeyType,
+    ) -> DatenLordResult<Arc<mpsc::Receiver<(String, Option<ValueType>)>>> {
+        // Create a mpsc channel, default capacity is 1024
+        let (tx, rx) = mpsc::channel(1024);
+
+        let mut client = self.client.clone();
+        // Try to watch the key prefix
+        let opt = etcd_client::WatchOptions::new().with_prefix();
+        let (_watcher, mut watch_stream) = client
+            .watch(prefix.to_string_key(), Some(opt.clone()))
+            .await
+            .with_context(|| "Failed to create watcher".to_owned())?;
+
+        let self_prefix = prefix.to_string_key();
+        // Spawn a new task to handle the watch stream
+        tokio::spawn(async move {
```

Comment:
This task should be taken over by the task manager.

Reply:
OK, I add it to the global task manager.